src/components/ContentHighlights/HighlightStepper/HighlightStepperConfirmContent.tsx

```diff
@@ -85,15 +95,29 @@ export const SelectedContent = ({ enterpriseId }) => {
 
   /* eslint-disable max-len */
   /**
-   * Results in a string like:
-   * `enterprise_customer_uuids:e783bb19-277f-479e-9c41-8b0ed31b4060 AND (aggregation_key:'course:edX+DemoX' OR aggregation_key:'course:edX+DemoX2')
+   * Results in a string like::
+   *   Secured Algolia API key (filters by catalog query uuids):
+   *   `(aggregation_key:'course:edX+DemoX' OR aggregation_key:'course:edX+DemoX2')`
+   *
+   *   Legacy:
+   *   `enterprise_customer_uuids:e783bb19-277f-479e-9c41-8b0ed31b4060 AND (aggregation_key:'course:edX+DemoX' OR aggregation_key:'course:edX+DemoX2')
    */
   /* eslint-enable max-len */
   const algoliaFilters = useMemo(() => {
+    if (isLoadingSecuredAlgoliaApiKey) {
+      // Don't show any results until the secured API key is loaded (if enabled)
+      return '';
+    }
+    let filterString = '';
+    if (!hasSecuredAlgoliaApiKey) {
+      filterString = `enterprise_customer_uuids:${ENABLE_TESTING(enterpriseId)}`;
+    }
     // import testEnterpriseId from the existing ../data/constants folder and replace with enterpriseId to test locally
-    let filterString = `enterprise_customer_uuids:${ENABLE_TESTING(enterpriseId)}`;
     if (currentSelectedRowIds.length > 0) {
-      filterString += ' AND (';
+      if (filterString) {
+        filterString += ' AND ';
+      }
```

Comment:
[curious] Should this filter string check be within the first conditional that encompasses the query were building here as opposed to just the ` AND  ` string? 
``` js
if (filterString && currentSelectedRowIds.length > 0) {
...
}
```

If `!!filterString` is false, the query returned could be `( selectedId1 OR selectedId2)` 

In that case should we just return the empty filter string?

Reply:
After async conversation, the return is what we expect with `( selectedId1 OR selectedId2 )` where the base filter is encoded into the algolia search key itself. The updated logic is in support of both an encoded algolia search key format and legacy.